### DISCUSSION TITLE
How to do polling with Inertia

### DISCUSSION THREAD

**Discussion by cwaidelich** [inertiajs/inertia#501] [Help (Vue)] [answered]
Hi,

how would you implement the polling functionality with inertia? Without Inertia, I would but a Intervall Function and fetch (some) data every x seconds.

I am working myself over the PING crm example. 
So, let´s say one user A changes a contact, how is the contact being updated on user B, when he is on the same page? Is this integrated in Inertia?

I hope you can understand the situation. 

Cheers

**Comment by cwaidelich** [comment, marked as answer]
Hello guys,

I just wanted to give you a short feedback on my own question, because this is very easy to do with inertia.

Taking into account how polling should be done in vue (by creating the SetInterval function, google for this or give feedback, if this is not best practice), the way to go is like so:

````
        methods: {
         updateData() {
            this.$inertia.reload({ preserveState: true , preserveScroll: true })  // read about the possible options under manual visits.
          },
        },
        created() {
          this.interval = setInterval(function () {
              this.updateData();
          }.bind(this), 1000);  // set 1000 to any number you need
        },
        beforeDestroy() {
            clearInterval(this.interval); // this is important!
        }
````

Link to [Manual Visits](https://inertiajs.com/manual-visits) for possible options.

I will be working in this a little bit more, so ask questions if you need some feedback.

Christian

**Comment by lostdesign** [comment]
To do what you wanna do, websockets might be the best way of dealing with this. E.g. emitting an event that this page has been updated by another user. Polling works fine as well, probably could do normal API polling with an JSON endpoint rather than fetching the same site.

Both work just fine, depends on what you wanna do.
